src/test/java/net/openhft/chronicle/queue/impl/single/TSQueueLockTest.java

```diff
@@ -0,0 +1,284 @@
+package net.openhft.chronicle.queue.impl.single;
+
+import net.openhft.chronicle.core.Jvm;
+import net.openhft.chronicle.core.io.Closeable;
+import net.openhft.chronicle.core.io.IOTools;
+import net.openhft.chronicle.core.onoes.LogLevel;
+import net.openhft.chronicle.core.threads.InterruptedRuntimeException;
+import net.openhft.chronicle.queue.QueueTestCommon;
+import net.openhft.chronicle.queue.impl.TableStore;
+import net.openhft.chronicle.queue.impl.table.Metadata;
+import net.openhft.chronicle.queue.impl.table.SingleTableBuilder;
+import net.openhft.chronicle.testframework.process.ProcessRunner;
+import net.openhft.chronicle.threads.Pauser;
+import net.openhft.chronicle.threads.Threads;
+import net.openhft.chronicle.wire.UnrecoverableTimeoutException;
+import org.jetbrains.annotations.NotNull;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.*;
+
+public class TSQueueLockTest extends QueueTestCommon {
```

Comment:
this is copy/pasted from TableStoreWriteLockTest. This test definitely needs some more work/refactoring - see https://github.com/OpenHFT/Chronicle-Queue/issues/994